src/wc.js

```diff
@@ -0,0 +1,159 @@
+//@ var counts = wc('file.txt'); //returns charCount lineCount wordCount
+//@ var charCount = wc('-m','file.txt'); //returns number of characters
+//@ var lineCount = wc('-l','file.txt'); //returns number of newlines
+//@ var wordCount = wc('-w','file.txt'); //returns number of words 
+//@ var countFiles = wc('file1', 'file2'); // returns: lineCount wordCount charCount 'file1'\nlineCount wordCount charCount 'file2' \n totalLines totalWords totalChars 'total'
+//@ var countFiles = wc(['file1', 'file2']); //same as above
+//@ ```
+//@
+//@ Read the end of a file.
+function _wc(options, files) {
+    var wc = [];
+    var pipe = common.readFromPipe();
+
+    if (!files && !pipe) common.error('no paths given');
+
+    var idx = 1;
+
+    /*  - from tail.js, which I'm using as a template. Not sure I need this section
+            - so each file will get the options applied to it?
+
+        if (options.numLines === true) {
+            idx = 2;
+            options.numLines = Number(arguments[1]);
+        } else if (options.numLines === false) {
+            options.numLines = 10;
+        }
+        options.numLines = -1 * Math.abs(options.numLines);
+
+    */
+
+    files = [].slice.call(arguments, idx);
+
+    if (pipe) {
+        files.unshift('-');
+    }
+
+    var moreThan1 = false;
+    var totalLines = 0;
+    var totalWords = 0;
+    var totalChars = 0;
+    if (files.length > 1) moreThan1 = true;
+    files.forEach(function(file) {
+        if (file !== '-') {
+            if (!fs.existsSync(file)) {
+                common.error('no such file or directory: ' + file, { continue: true });
+                return;
+            } else if (common.statFollowLinks(file).isDirectory()) {
+                common.error("error reading '" + file + "': Is a directory", {
+                    continue: true,
+                });
+                return;
+            }
+        }
+
+        var contents = file === '-' ? pipe : fs.readFileSync(file, 'utf8');
+        /*
+        'm': 'charCount',
+        'l': 'lineCount',
+        'w': 'wordCount',
+        */
+
+        var temp = '';
+        var thisLines = getLines(contents);
+        var thisWords = getWords(contents);
+        var thisChars = getChars(contents);
+        totalLines += parseInt(thisLines);
+        totalWords += parseInt(thisWords);
+        totalChars += parseInt(thisChars);
+
+        if (options.charCount || options.lineCount || options.wordCount) {
+            //modify wc accordingly
+
+            if (options.lineCount) temp += `${thisLines} `;
+            if (options.wordCount) temp += `${thisWords} `;
+            if (options.charCount) temp += `${thisChars} `;
+        } else {
+            //get all variables! 
+            temp += `${thisLines} ${thisWords} ${thisChars}`;
+        }
+        temp += ` ${file}`
+
+        wc.push(temp);
+
+    });
+
+    if (moreThan1) wc.push(`${totalLines} ${totalWords} ${totalChars} total`);
+
+    return wc.join('\n');
```

Comment:
We probably want to end this with a trailing newline. I think most other commands follow this pattern.